src/control/server/mgmt_system.go

```diff
@@ -348,6 +349,72 @@ func (svc *mgmtSvc) PingRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmtp
 	return resp, nil
 }
 
+// ReformatRanks implements the method defined for the Management Service.
+//
+// Reformat storage of data-plane instances (DAOS system members) managed by harness.
+func (svc *mgmtSvc) ReformatRanks(parent context.Context, req *mgmtpb.RanksReq) (*mgmtpb.RanksResp, error) {
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+	if len(req.GetRanks()) == 0 {
+		return nil, errors.New("no ranks specified in request")
+	}
+	svc.log.Debugf("MgmtSvc.ReformatRanks dispatch, req:%+v\n", *req)
+
+	instances, err := svc.localInstances(req.GetRanks())
+	if err != nil {
+		return nil, err
+	}
+	for _, srv := range instances {
+		if srv.isStarted() {
+			return nil, FaultInstancesNotStopped("reformat", srv.Index())
+		}
+		if err := srv.RemoveSuperblock(); err != nil {
+			return nil, err
+		}
+		srv.startLoop <- true // proceed to awaiting storage format
+	}
+
+	ready := make(chan struct{})
+	go func() {
+		for {
+			success := true
+			for _, srv := range instances {
+				if srv.waitStorage.IsFalse() {
+					success = false
+				}
+			}
+			if !success {
+				time.Sleep(instanceUpdateDelay)
+				continue
+			}
+			close(ready)
+			break
+		}
+	}()
+
+	select {
+	case <-ready: // all relevant instances are waiting for storage format
+		svc.log.Debug("MgmtSvc.ReformatRanks instances waiting for format")
+	case <-time.After(svc.harness.rankStartTimeout):
+		svc.log.Debug("MgmtSvc.ReformatRanks timeout exceeded")
+	}
```

Comment:
Should add a case for `<-parent.Done()`, right?

Reply:
yes that is missing